# <p><center> **GUIA 1**</center></p>

## <p><center> **ACTIVIDAD 1**</center></p>  
### Reconocer los tipos de datos de las variables con las que se cuenta y aplicar las funciones correspondientes a ellas.

##### cargue y lectura del df

In [29]:
#Importación de la librería pandas
import pandas as pd

In [17]:
#lectura de la base de datos
df = pd.read_csv("census_income.csv")

##### Caracterización inicial del df.  
Se realizan una caracterización inicial en la que se identifica cantidad de filas, columnas, tipos de variables (enteros, flotantes, string y valoes no nulos) y sus respectivos nombres. tambien, en las variables numericas usando el método describe se identifican descriptores como conte, media, entre otros.

In [53]:
#conteo de las filas y columnas dentro de la variable
df.shape

(48842, 16)

In [9]:
#cantidad de columnas
df.shape[1]

16

In [11]:
# Cantidad de filas
len(df)

48842

In [13]:
#Verificación de las columnas con el nombre de la variable
df.columns

Index(['Unnamed: 0', 'age', 'workclass', 'fnlwgt', 'education',
       'education-num', 'marital-status', 'occupation', 'relationship', 'race',
       'sex', 'capital-gain', 'capital-loss', 'hours-per-week',
       'native-country', 'income'],
      dtype='object')

In [276]:
# Verificación inicial del contenido de la base de datos, con este metodo traigo las primeras cinco filas de la base de datos.
df.head(5)

Unnamed: 0  age         workclass  fnlwgt  education  education-num  \
0           0   39         State-gov   77516  Bachelors             13   
1           1   50  Self-emp-not-inc   83311  Bachelors             13   
2           2   38           Private  215646    HS-grad              9   
3           3   53           Private  234721       11th              7   
4           4   28           Private  338409  Bachelors             13   

       marital-status         occupation   relationship   race     sex  \
0       Never-married       Adm-clerical  Not-in-family  White    Male   
1  Married-civ-spouse    Exec-managerial        Husband  White    Male   
2            Divorced  Handlers-cleaners  Not-in-family  White    Male   
3  Married-civ-spouse  Handlers-cleaners        Husband  Black    Male   
4  Married-civ-spouse     Prof-specialty           Wife  Black  Female   

   capital-gain  capital-loss  hours-per-week native-country income  
0          2174             0              40  United-States  <=50K  
1             0             0              13  United-States  <=50K  
2             0             0              40  United-States  <=50K  
3             0             0              40  United-States  <=50K  
4             0             0              40           Cuba  <=50K

In [19]:
#información de cada una de las variables en cuanto a la cantidad de datos y tipo de datos en cada una de ellas
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48842 entries, 0 to 48841
Data columns (total 16 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   Unnamed: 0      48842 non-null  int64 
 1   age             48842 non-null  int64 
 2   workclass       47879 non-null  object
 3   fnlwgt          48842 non-null  int64 
 4   education       48842 non-null  object
 5   education-num   48842 non-null  int64 
 6   marital-status  48842 non-null  object
 7   occupation      47876 non-null  object
 8   relationship    48842 non-null  object
 9   race            48842 non-null  object
 10  sex             48842 non-null  object
 11  capital-gain    48842 non-null  int64 
 12  capital-loss    48842 non-null  int64 
 13  hours-per-week  48842 non-null  int64 
 14  native-country  48568 non-null  object
 15  income          48842 non-null  object
dtypes: int64(7), object(9)
memory usage: 6.0+ MB


In [109]:
#metodo en el cuál solamente se tienen en cuenta las variables cuantitativas 
#y se realizan diversos cálculos, como conteo, media, desviación, minimo, maximo, percentiles.
df.describe()

Unnamed: 0           age        fnlwgt  education-num  capital-gain  \
count  48842.000000  48842.000000  4.884200e+04   48842.000000  48842.000000   
mean   24420.500000     38.643585  1.896641e+05      10.078089   1079.067626   
std    14099.615261     13.710510  1.056040e+05       2.570973   7452.019058   
min        0.000000     17.000000  1.228500e+04       1.000000      0.000000   
25%    12210.250000     28.000000  1.175505e+05       9.000000      0.000000   
50%    24420.500000     37.000000  1.781445e+05      10.000000      0.000000   
75%    36630.750000     48.000000  2.376420e+05      12.000000      0.000000   
max    48841.000000     90.000000  1.490400e+06      16.000000  99999.000000   

       capital-loss  hours-per-week  
count  48842.000000    48842.000000  
mean      87.502314       40.422382  
std      403.004552       12.391444  
min        0.000000        1.000000  
25%        0.000000       40.000000  
50%        0.000000       40.000000  
75%        0.000000       45.000000  
max     4356.000000       99.000000

In [280]:
# Traigo 10 filas al azar
df.sample(10)

Unnamed: 0  age    workclass  fnlwgt     education  education-num  \
23470       23470   27      Private  271603       7th-8th              4   
40071       40071   33      Private  170651     Bachelors             13   
39831       39831   36          NaN   92440  Some-college             10   
22970       22970   51  Federal-gov   20795       HS-grad              9   
3790         3790   47      Private  140664     Bachelors             13   
37899       37899   18      Private  106943          11th              7   
20140       20140   47      Private  198769  Some-college             10   
5578         5578   29      Private  290740    Assoc-acdm             12   
14493       14493   34      Private  172579     Assoc-voc             11   
28548       28548   68            ?  186266       HS-grad              9   

           marital-status        occupation   relationship   race     sex  \
23470       Never-married     Other-service  Not-in-family  White    Male   
40071  Married-civ-spouse   Exec-managerial           Wife  White  Female   
39831  Married-civ-spouse               NaN        Husband  White    Male   
22970  Married-civ-spouse  Transport-moving        Husband  White    Male   
3790             Divorced             Sales  Not-in-family  White    Male   
37899       Never-married             Sales      Own-child  White  Female   
20140       Never-married      Adm-clerical  Not-in-family  White  Female   
5578        Never-married   Priv-house-serv  Not-in-family  White  Female   
14493            Divorced      Tech-support      Unmarried  White  Female   
28548  Married-civ-spouse                 ?           Wife  White  Female   

       capital-gain  capital-loss  hours-per-week native-country  income  
23470             0             0              24              ?   <=50K  
40071             0          1977              38  United-States   >50K.  
39831             0             0              40  United-States  <=50K.  
22970          7688             0              40  United-States    >50K  
3790              0             0              45  United-States   <=50K  
37899             0             0              12  United-States  <=50K.  
20140             0             0              40  United-States    >50K  
5578              0             0              50  United-States   <=50K  
14493             0             0              40  United-States   <=50K  
28548             0             0               8  United-States   <=50K

##### Unnamed  
Corresponde a una variable que muestra un consecutivo, por lo que no le realizo ningún análisis

##### Age  
Corresponde a una variable cuantitativa, de tipo entero,se encuentran datos desde los 17 años como edad mínima, hasta los 90 como edad máxima, cuenta con 74 datos únicos, con una media de 38.6, siendo 36 años la edad que más se repite. La variable no cuenta con datos nulos. 

In [105]:
#información de la variable en cuanto a la cantidad de datos y tipo de dato.
df["age"].info()

<class 'pandas.core.series.Series'>
RangeIndex: 48842 entries, 0 to 48841
Series name: age
Non-Null Count  Dtype
--------------  -----
48842 non-null  int64
dtypes: int64(1)
memory usage: 381.7 KB


In [111]:
#Metodo con el cual se realizan cálculos estadisticos, como conteo, media y otros. solamente aplica en variables cuantitativas.
df["age"].describe()

count    48842.000000
mean        38.643585
std         13.710510
min         17.000000
25%         28.000000
50%         37.000000
75%         48.000000
max         90.000000
Name: age, dtype: float64

In [63]:
# Metodo para identificar el número de registros en la variable
df["age"].count()

48842

In [282]:
#Método con el cual se realiza el conteo de cada uno de los valores dentro de la variable
df["age"].value_counts()

age
36    1348
35    1337
33    1335
23    1329
31    1325
      ... 
88       6
85       5
87       3
89       2
86       1
Name: count, Length: 74, dtype: int64

In [89]:
# Metodo para identificar el número más grande dentro de la variable
df["age"].max()

90

In [91]:
# Metodo para identificar el número más pequeño dentro de la variable
df["age"].min()

17

In [93]:
# Metodo para identificar el valor que aparece con mayor frecuencia dentro de la variable
df["age"].mode()

0    36
Name: age, dtype: int64

In [284]:
# Metodo para identificar el promedio dentro de la variable
df["age"].mean()

38.64358543876172

In [95]:
# Metodo para identificar el número de valores únicos dentro de una variable
df["age"].nunique()

74

In [113]:
# Metodo para contar los valores nulos
df["age"].isnull().sum()

0

##### workclass  
Es una variable cualitativa correspondiente a clase trabajadora, es de tipo string, cuenta con 9 categorias, de las cuales la que más se repite es _Private_, con 33906 registros y la que menos se repite es _Never- worked_ con 10. En cuanto a los valores nulos se encontraron 963 registros y 47879 no nulos. Adicionalmente, se resalta que existe una categoría con el simbolo "?", con 1836 registros, que tampoco nos brinda mayor información en comparación con los valores nulos.

In [334]:
#información de la variable en cuanto a la cantidad de datos y tipo de dato.
df["workclass"].info()

<class 'pandas.core.series.Series'>
RangeIndex: 48842 entries, 0 to 48841
Series name: workclass
Non-Null Count  Dtype 
--------------  ----- 
47879 non-null  object
dtypes: object(1)
memory usage: 381.7+ KB


In [99]:
df["workclass"].describe()

count       47879
unique          9
top       Private
freq        33906
Name: workclass, dtype: object

In [336]:
# Metodo para identificar el número de registros en la variable
df["age"].count()

48842

In [165]:
#Frecuencia de cada una de las opciones de la variable
print(df["workclass"].value_counts())
df["workclass"].value_counts().sum() ### Con este método me doy cuenta que no están todos los datos, de acuerdo con el número de filas. 

workclass
Private             33906
Self-emp-not-inc     3862
Local-gov            3136
State-gov            1981
?                    1836
Self-emp-inc         1695
Federal-gov          1432
Without-pay            21
Never-worked           10
Name: count, dtype: int64


47879

In [109]:
# Nombres de las categorías dentro de la variable
df["workclass"].nunique()

9

In [107]:
# Nombres de las categorías dentro de la variable
pd.unique(df["workclass"])

array(['State-gov', 'Self-emp-not-inc', 'Private', 'Federal-gov',
       'Local-gov', '?', 'Self-emp-inc', 'Without-pay', 'Never-worked',
       nan], dtype=object)

In [157]:
#conteo de los valores nulos dentro de la variable
df["workclass"].isnull().sum() 

963

##### fnlwgt
Es una varibale de tipo entero, que cuenta con 48842 registros con 28523 datos únicos, de los cuales el valor máximo es de 1490400 y el mínimo 12285, el valor que más se repite es 203488 con 21 registros, el promedio es de 189664. No cuenta con valores nulos. 

In [146]:
df["fnlwgt"].info()

<class 'pandas.core.series.Series'>
RangeIndex: 48842 entries, 0 to 48841
Series name: fnlwgt
Non-Null Count  Dtype
--------------  -----
48842 non-null  int64
dtypes: int64(1)
memory usage: 381.7 KB


In [339]:
#Metodo con el cual se realizan cálculos estadisticos, como conteo, media y otros. solamente aplica en variables cuantitativas.
df["fnlwgt"].describe()

count    4.884200e+04
mean     1.896641e+05
std      1.056040e+05
min      1.228500e+04
25%      1.175505e+05
50%      1.781445e+05
75%      2.376420e+05
max      1.490400e+06
Name: fnlwgt, dtype: float64

In [341]:
# Metodo para identificar el número de registros en la variable
df["fnlwgt"].count()

48842

In [154]:
#Frecuencia de cada una de las opciones de la variable
df["fnlwgt"].value_counts()

fnlwgt
203488    21
120277    19
190290    19
125892    18
126569    18
          ..
286983     1
185942     1
234220     1
214706     1
350977     1
Name: count, Length: 28523, dtype: int64

In [164]:
df["fnlwgt"].max()

1490400

In [166]:
df[df.fnlwgt == 1490400]

Unnamed: 0  age workclass   fnlwgt     education  education-num  \
40535       40535   27   Private  1490400  Some-college             10   

           marital-status       occupation relationship   race   sex  \
40535  Married-civ-spouse  Exec-managerial      Husband  White  Male   

       capital-gain  capital-loss  hours-per-week native-country  income  
40535             0             0              40  United-States  <=50K.

In [162]:
df["fnlwgt"].min()

12285

In [168]:
df[df.fnlwgt == 12285]

Unnamed: 0  age workclass  fnlwgt     education  education-num  \
17096       17096   25         ?   12285  Some-college             10   

      marital-status occupation   relationship                race     sex  \
17096  Never-married          ?  Not-in-family  Amer-Indian-Eskimo  Female   

       capital-gain  capital-loss  hours-per-week native-country income  
17096             0             0              20  United-States  <=50K

In [170]:
df["fnlwgt"].mode()

0    203488
Name: fnlwgt, dtype: int64

In [198]:
###Identificación de los registros que mas se repiten 
print(df[df.fnlwgt == 203488])
###conteo de los registros que mas se repiten 
print(df[df.fnlwgt == 203488].count())

       Unnamed: 0  age         workclass  fnlwgt     education  education-num  \
3464         3464   31           Private  203488     Bachelors             13   
9023         9023   30           Private  203488  Some-college             10   
10383       10383   33           Private  203488  Some-college             10   
10766       10766   34           Private  203488  Some-college             10   
17553       17553   30           Private  203488  Some-college             10   
17793       17793   31           Private  203488       HS-grad              9   
18916       18916   31      Self-emp-inc  203488     Bachelors             13   
20313       20313   33           Private  203488  Some-college             10   
22523       22523   34           Private  203488       HS-grad              9   
22723       22723   34           Private  203488     Bachelors             13   
22813       22813   31       Federal-gov  203488    Assoc-acdm             12   
23625       23625   34  Self

In [183]:
df["fnlwgt"].mean()

189664.13459727284

In [186]:
df["fnlwgt"].nunique()

28523

In [115]:
df["fnlwgt"].isnull().sum()

0

##### education  
Variable cualitativa, de tipo string, ordinal, cuenta con 48842 registros y 16 categorías, de las cuales la que más se presenta es _HS-grad_ con 15784 registros y con 83 registros _Preschool_ es la que menos se repite, la variable no cuenta con datos nulos.

In [194]:
df["education"].describe()

count       48842
unique         16
top       HS-grad
freq        15784
Name: education, dtype: object

In [346]:
df["education"].info()

<class 'pandas.core.series.Series'>
RangeIndex: 48842 entries, 0 to 48841
Series name: education
Non-Null Count  Dtype 
--------------  ----- 
48842 non-null  object
dtypes: object(1)
memory usage: 381.7+ KB


In [188]:
#Frecuencia de cada una de las opciones de la variable
df["education"].count()

48842

In [95]:
#Frecuencia de cada una de las opciones de la variable
df["education"].value_counts()

education
HS-grad         15784
Some-college    10878
Bachelors        8025
Masters          2657
Assoc-voc        2061
11th             1812
Assoc-acdm       1601
10th             1389
7th-8th           955
Prof-school       834
9th               756
12th              657
Doctorate         594
5th-6th           509
1st-4th           247
Preschool          83
Name: count, dtype: int64

In [203]:
df["education"].nunique()

16

In [205]:
pd.unique(df["education"])

array(['Bachelors', 'HS-grad', '11th', 'Masters', '9th', 'Some-college',
       'Assoc-acdm', 'Assoc-voc', '7th-8th', 'Doctorate', 'Prof-school',
       '5th-6th', '10th', '1st-4th', 'Preschool', '12th'], dtype=object)

In [117]:
df["education"].isnull().sum()

0

##### education-num  
Variable cualitativa, de tipo entero, ordinal, cuenta con 48842 registros y 16 categorías, de las cuales la que más se presenta es _**9**_ con 15784 registros y con 83 registros _**1**_ es la que menos se repite, la variable no cuenta con datos nulos.

In [348]:
#información de la variable en cuanto a la cantidad de datos y tipo de dato.
df["education-num"].info()

<class 'pandas.core.series.Series'>
RangeIndex: 48842 entries, 0 to 48841
Series name: education-num
Non-Null Count  Dtype
--------------  -----
48842 non-null  int64
dtypes: int64(1)
memory usage: 381.7 KB


In [350]:
#Metodo con el cual se realizan cálculos estadisticos, como conteo, media y otros. solamente aplica en variables cuantitativas.
df["education-num"].describe()

count    48842.000000
mean        10.078089
std          2.570973
min          1.000000
25%          9.000000
50%         10.000000
75%         12.000000
max         16.000000
Name: education-num, dtype: float64

In [352]:
# Metodo para identificar el número de registros en la variable
df["education-num"].count()

48842

In [354]:
#Frecuencia de cada una de las opciones de la variable
df["education-num"].value_counts()

education-num
9     15784
10    10878
13     8025
14     2657
11     2061
7      1812
12     1601
6      1389
4       955
15      834
5       756
8       657
16      594
3       509
2       247
1        83
Name: count, dtype: int64

In [356]:
# Metodo para identificar el valor que aparece con mayor frecuencia dentro de la variable
df["education-num"].mode()

0    9
Name: education-num, dtype: int64

In [358]:
# Metodo para identificar el número de valores únicos dentro de una variable
df["education-num"].nunique()

16

In [360]:
# Nombres de las categorías dentro de la variable
pd.unique(df["education-num"])

array([13,  9,  7, 14,  5, 10, 12, 11,  4, 16, 15,  3,  6,  2,  1,  8],
      dtype=int64)

In [362]:
df["education-num"].isnull().sum()

0

##### marital-status
Es una variable cualitativa correspondiente a estado civil, es de tipo string, tiene 48842 registros, divididos en 7 categorias, de las cuales la que más se repite es _Married-civ-spouse_ con 22379 registros y la que menos se repite es _Married-AF-spouse_ con 37. En cuanto a los valores nulos no se encontraron registros.

In [215]:
df["marital-status"].info()

<class 'pandas.core.series.Series'>
RangeIndex: 48842 entries, 0 to 48841
Series name: marital-status
Non-Null Count  Dtype 
--------------  ----- 
48842 non-null  object
dtypes: object(1)
memory usage: 381.7+ KB


In [217]:
df["marital-status"].describe()

count                  48842
unique                     7
top       Married-civ-spouse
freq                   22379
Name: marital-status, dtype: object

In [213]:
df["marital-status"].count()

48842

In [85]:
#Frecuencia de cada una de las opciones de la variable
df["marital-status"].value_counts()

marital-status
Married-civ-spouse       22379
Never-married            16117
Divorced                  6633
Separated                 1530
Widowed                   1518
Married-spouse-absent      628
Married-AF-spouse           37
Name: count, dtype: int64

In [220]:
df["marital-status"].nunique()

7

In [240]:
pd.unique(df["marital-status"])

array(['Never-married', 'Married-civ-spouse', 'Divorced',
       'Married-spouse-absent', 'Separated', 'Married-AF-spouse',
       'Widowed'], dtype=object)

In [121]:
df["marital-status"].isnull().sum()

0

##### occupation  
Es una variable cualitativa, nominal, correspondiente a ocupación, es de tipo string, tiene 47876 registros no nulos de los 48842 de la base de datos , está dividida en 15 categorias, de las cuales la que más se repite es _Prof-specialty_ con 6172 registros y la que menos se repite es _Armed-Forces_ con 15. En cuanto a los valores nulos se encontraron 996 registros y dentro de las categorías, se encuentra con 1843 registros el simbolo _**?**_ siendo un registro no válido.

In [234]:
df["occupation"].info()

<class 'pandas.core.series.Series'>
RangeIndex: 48842 entries, 0 to 48841
Series name: occupation
Non-Null Count  Dtype 
--------------  ----- 
47876 non-null  object
dtypes: object(1)
memory usage: 381.7+ KB


In [236]:
df["occupation"].describe()

count              47876
unique                15
top       Prof-specialty
freq                6172
Name: occupation, dtype: object

In [226]:
df["occupation"].count()

47876

In [83]:
#Frecuencia de cada una de las opciones de la variable
df["occupation"].value_counts()

occupation
Prof-specialty       6172
Craft-repair         6112
Exec-managerial      6086
Adm-clerical         5611
Sales                5504
Other-service        4923
Machine-op-inspct    3022
Transport-moving     2355
Handlers-cleaners    2072
?                    1843
Farming-fishing      1490
Tech-support         1446
Protective-serv       983
Priv-house-serv       242
Armed-Forces           15
Name: count, dtype: int64

In [238]:
df["occupation"].nunique()

15

In [242]:
pd.unique(df["occupation"])

array(['Adm-clerical', 'Exec-managerial', 'Handlers-cleaners',
       'Prof-specialty', 'Other-service', 'Sales', 'Craft-repair',
       'Transport-moving', 'Farming-fishing', 'Machine-op-inspct',
       'Tech-support', '?', 'Protective-serv', 'Armed-Forces',
       'Priv-house-serv', nan], dtype=object)

In [167]:
#conteo de los valores nulos dentro de la variable
df["occupation"].isnull().sum() 

966

##### relationship  
Es una variable cualitativa, nominal, correspondiente a Relacion, es de tipo string, tiene 48842 registros cuenta con 6 categorias, de las cuales la que más se repite es _Husband_, con 19716 registros y la que menos se repite es _Other-relative_ con 1506. En cuanto a los valores nulos no se encontraron registros.

In [254]:
df["relationship"].info()

<class 'pandas.core.series.Series'>
RangeIndex: 48842 entries, 0 to 48841
Series name: relationship
Non-Null Count  Dtype 
--------------  ----- 
48842 non-null  object
dtypes: object(1)
memory usage: 381.7+ KB


In [251]:
df["relationship"].describe()

count       48842
unique          6
top       Husband
freq        19716
Name: relationship, dtype: object

In [247]:
df["relationship"].count()

48842

In [28]:
#Frecuencia de cada una de las opciones de la variable
df["relationship"].value_counts()

relationship
Husband           19716
Not-in-family     12583
Own-child          7581
Unmarried          5125
Wife               2331
Other-relative     1506
Name: count, dtype: int64

In [260]:
df["relationship"].nunique()

6

In [258]:
pd.unique(df["marital-status"])

array(['Never-married', 'Married-civ-spouse', 'Divorced',
       'Married-spouse-absent', 'Separated', 'Married-AF-spouse',
       'Widowed'], dtype=object)

In [123]:
df["relationship"].isnull().sum()

0

##### race
Es una variable cualitativa, nominal correspondiente a raza, es de tipo string, cuenta con 5 categorias, de las cuales la que más se repite es _White_, con 41762 registros y la que menos se repite es _Other_ con 406. En cuanto a los valores nulos no se encontraron registros.

In [262]:
df["race"].info()

<class 'pandas.core.series.Series'>
RangeIndex: 48842 entries, 0 to 48841
Series name: race
Non-Null Count  Dtype 
--------------  ----- 
48842 non-null  object
dtypes: object(1)
memory usage: 381.7+ KB


In [264]:
df["race"].describe()

count     48842
unique        5
top       White
freq      41762
Name: race, dtype: object

In [270]:
df["race"].count()

48842

In [266]:
df["race"].value_counts()

race
White                 41762
Black                  4685
Asian-Pac-Islander     1519
Amer-Indian-Eskimo      470
Other                   406
Name: count, dtype: int64

In [272]:
df["race"].nunique()

5

In [274]:
pd.unique(df["race"])

array(['White', 'Black', 'Asian-Pac-Islander', 'Amer-Indian-Eskimo',
       'Other'], dtype=object)

In [125]:
df["race"].isnull().sum()

0

##### sex
Es una variable cualitativa correspondiente a sexo, es de tipo string, cuenta con 2 categorias, de las cuales la que más se repite es _Male_, con 32650 registros. En cuanto a los valores nulos no se encontraron registros.

In [370]:
#información de la variable en cuanto a la cantidad de datos y tipo de dato.
df["sex"].info()

<class 'pandas.core.series.Series'>
RangeIndex: 48842 entries, 0 to 48841
Series name: age
Non-Null Count  Dtype
--------------  -----
48842 non-null  int64
dtypes: int64(1)
memory usage: 381.7 KB


In [374]:
#Metodo con el cual se realizan cálculos estadisticos, como conteo, media y otros. solamente aplica en variables cuantitativas.
df["sex"].describe()

count     48842
unique        2
top        Male
freq      32650
Name: sex, dtype: object

In [377]:
# Metodo para identificar el número de registros en la variable
df["sex"].count()

48842

In [381]:
#Método con el cual se realiza el conteo de cada uno de los valores dentro de la variable
df["sex"].value_counts()

sex
Male      32650
Female    16192
Name: count, dtype: int64

In [385]:
# Metodo para identificar el número de valores únicos dentro de una variable
df["sex"].nunique()

2

In [389]:
# Nombres de las categorías dentro de la variable
pd.unique(df["sex"])

array(['Male', 'Female'], dtype=object)

In [127]:
df["sex"].isnull().sum()

0

##### capital-gain
Es una variable cuantitativa, de tipo entero, cuenta con 48842 registros con un rango entre 0 y 99999. con 123 datos únicos. en la mayoría de los casos (44807) el valor fue de cero. la variable no contiene datos nulos.

In [392]:
#información de la variable en cuanto a la cantidad de datos y tipo de dato.
df["capital-gain"].info()

<class 'pandas.core.series.Series'>
RangeIndex: 48842 entries, 0 to 48841
Series name: age
Non-Null Count  Dtype
--------------  -----
48842 non-null  int64
dtypes: int64(1)
memory usage: 381.7 KB


In [410]:
#Metodo con el cual se realizan cálculos estadisticos, como conteo, media y otros. solamente aplica en variables cuantitativas.
df["capital-gain"].describe()

count    48842.000000
mean      1079.067626
std       7452.019058
min          0.000000
25%          0.000000
50%          0.000000
75%          0.000000
max      99999.000000
Name: capital-gain, dtype: float64

In [412]:
# Metodo para identificar el número de registros en la variable
df["capital-gain"].count()

48842

In [414]:
#Método con el cual se realiza el conteo de cada uno de los valores dentro de la variable
df["capital-gain"].value_counts()

capital-gain
0        44807
15024      513
7688       410
7298       364
99999      244
         ...  
22040        1
2387         1
1639         1
1111         1
6612         1
Name: count, Length: 123, dtype: int64

In [400]:
# Metodo para identificar el número más grande dentro de la variable
df["capital-gain"].max()

99999

In [402]:
# Metodo para identificar el número más pequeño dentro de la variable
df["capital-gain"].min()

0

In [404]:
# Metodo para identificar el valor que aparece con mayor frecuencia dentro de la variable
df["capital-gain"].mode()

0    0
Name: capital-gain, dtype: int64

In [406]:
# Metodo para identificar el número de valores únicos dentro de una variable
df["capital-gain"].nunique()

123

In [408]:
# Metodo para contar los valores nulos
df["capital-gain"].isnull().sum()

0

##### capital-loss
Es una variable cuantitativa, de tipo entero, cuenta con 48842 registros con un rango entre 0 y 4356. con 99 datos únicos. en la mayoría de los casos (46560) el valor fue de cero. la variable no contiene datos nulos.

In [418]:
#información de la variable en cuanto a la cantidad de datos y tipo de dato.
df["capital-loss"].info()

<class 'pandas.core.series.Series'>
RangeIndex: 48842 entries, 0 to 48841
Series name: capital-loss
Non-Null Count  Dtype
--------------  -----
48842 non-null  int64
dtypes: int64(1)
memory usage: 381.7 KB


In [420]:
#Metodo con el cual se realizan cálculos estadisticos, como conteo, media y otros. solamente aplica en variables cuantitativas.
df["capital-loss"].describe()

count    48842.000000
mean        87.502314
std        403.004552
min          0.000000
25%          0.000000
50%          0.000000
75%          0.000000
max       4356.000000
Name: capital-loss, dtype: float64

In [422]:
# Metodo para identificar el número de registros en la variable
df["capital-loss"].count()

48842

In [424]:
#Método con el cual se realiza el conteo de cada uno de los valores dentro de la variable
df["capital-loss"].value_counts()

capital-loss
0       46560
1902      304
1977      253
1887      233
2415       72
        ...  
1539        1
1870        1
1911        1
2465        1
1421        1
Name: count, Length: 99, dtype: int64

In [426]:
# Metodo para identificar el número más grande dentro de la variable
df["capital-loss"].max()

4356

In [428]:
# Metodo para identificar el número más pequeño dentro de la variable
df["capital-loss"].min()

0

In [430]:
# Metodo para identificar el valor que aparece con mayor frecuencia dentro de la variable
df["capital-loss"].mode()

0    0
Name: capital-loss, dtype: int64

In [432]:
# Metodo para identificar el número de valores únicos dentro de una variable
df["capital-loss"].nunique()

99

In [434]:
# Nombres de las categorías dentro de la variable
pd.unique(df["capital-loss"])

array([   0, 2042, 1408, 1902, 1573, 1887, 1719, 1762, 1564, 2179, 1816,
       1980, 1977, 1876, 1340, 2206, 1741, 1485, 2339, 2415, 1380, 1721,
       2051, 2377, 1669, 2352, 1672,  653, 2392, 1504, 2001, 1590, 1651,
       1628, 1848, 1740, 2002, 1579, 2258, 1602,  419, 2547, 2174, 2205,
       1726, 2444, 1138, 2238,  625,  213, 1539,  880, 1668, 1092, 1594,
       3004, 2231, 1844,  810, 2824, 2559, 2057, 1974,  974, 2149, 1825,
       1735, 1258, 2129, 2603, 2282,  323, 4356, 2246, 1617, 1648, 2489,
       3770, 1755, 3683, 2267, 2080, 2457,  155, 3900, 2201, 1944, 2467,
       2163, 2754, 2472, 1411, 1429, 3175, 1510, 1870, 1911, 2465, 1421],
      dtype=int64)

In [436]:
df["capital-loss"].isnull().sum()

0

##### hours-per-week
ES una variable cuantitativa, de tipo entero, cuenta con 48842 registros con un rango entre 1 y 99. siendo 40 el número que más se repite con 22803 registros, la variable no contiene datos nulos.

In [441]:
#información de la variable en cuanto a la cantidad de datos y tipo de dato.
df["hours-per-week"].info()

<class 'pandas.core.series.Series'>
RangeIndex: 48842 entries, 0 to 48841
Series name: hours-per-week
Non-Null Count  Dtype
--------------  -----
48842 non-null  int64
dtypes: int64(1)
memory usage: 381.7 KB


In [443]:
#Metodo con el cual se realizan cálculos estadisticos, como conteo, media y otros. solamente aplica en variables cuantitativas.
df["hours-per-week"].describe()

count    48842.000000
mean        40.422382
std         12.391444
min          1.000000
25%         40.000000
50%         40.000000
75%         45.000000
max         99.000000
Name: hours-per-week, dtype: float64

In [445]:
# Metodo para identificar el número de registros en la variable
df["hours-per-week"].count()

48842

In [447]:
#Método con el cual se realiza el conteo de cada uno de los valores dentro de la variable
df["hours-per-week"].value_counts()

hours-per-week
40    22803
50     4246
45     2717
60     2177
35     1937
      ...  
79        1
94        1
82        1
87        1
69        1
Name: count, Length: 96, dtype: int64

In [449]:
# Metodo para identificar el número más grande dentro de la variable
df["hours-per-week"].max()

99

In [451]:
# Metodo para identificar el número más pequeño dentro de la variable
df["hours-per-week"].min()

1

In [453]:
# Metodo para identificar el valor que aparece con mayor frecuencia dentro de la variable
df["hours-per-week"].mode()

0    40
Name: hours-per-week, dtype: int64

In [455]:
# Metodo para identificar el número de valores únicos dentro de una variable
df["hours-per-week"].nunique()

96

In [469]:
# Nombres de las categorías dentro de la variable
pd.unique(df["hours-per-week"])

array([40, 13, 16, 45, 50, 80, 30, 35, 60, 20, 52, 44, 15, 25, 38, 43, 55,
       48, 58, 32, 70,  2, 22, 56, 41, 28, 36, 24, 46, 42, 12, 65,  1, 10,
       34, 75, 98, 33, 54,  8,  6, 64, 19, 18, 72,  5,  9, 47, 37, 21, 26,
       14,  4, 59,  7, 99, 53, 39, 62, 57, 78, 90, 66, 11, 49, 84,  3, 17,
       68, 27, 85, 31, 51, 77, 63, 23, 87, 88, 73, 89, 97, 94, 29, 96, 67,
       82, 86, 91, 81, 76, 92, 61, 74, 95, 79, 69], dtype=int64)

In [471]:
# Metodo para contar los valores nulos
df["hours-per-week"].isnull().sum()

0

##### native-country
Es una variable cualitativa correspondiente a pais nativo, es de tipo string, cuenta con 42 categorias, de las cuales la que más se repite es _United-States_, con 43832 registros y la que menos se repite es _Holand-Netherlands_ con 1. En cuanto a los valores nulos se encontraron 274 registros y 48568 no nulos. Adicionalmente, se resalta que existe una categoría con el simbolo _**?**_, con 583 registros, que tampoco nos brinda mayor información en comparación con los valores nulos.

In [312]:
#información de la variable en cuanto a la cantidad de datos y tipo de dato.
df["native-country"].info()

<class 'pandas.core.series.Series'>
RangeIndex: 48842 entries, 0 to 48841
Series name: native-country
Non-Null Count  Dtype 
--------------  ----- 
48568 non-null  object
dtypes: object(1)
memory usage: 381.7+ KB


In [314]:
#Metodo con el cual se realizan cálculos estadisticos, como conteo, media y otros. solamente aplica en variables cuantitativas.
df["native-country"].describe()

count             48568
unique               42
top       United-States
freq              43832
Name: native-country, dtype: object

In [316]:
# Metodo para identificar el número de registros en la variable
df["native-country"].count()

48568

In [318]:
#Método con el cual se realiza el conteo de cada uno de los valores dentro de la variable
df["native-country"].value_counts()

native-country
United-States                 43832
Mexico                          951
?                               583
Philippines                     295
Germany                         206
Puerto-Rico                     184
Canada                          182
El-Salvador                     155
India                           151
Cuba                            138
England                         127
China                           122
South                           115
Jamaica                         106
Italy                           105
Dominican-Republic              103
Japan                            92
Guatemala                        88
Poland                           87
Vietnam                          86
Columbia                         85
Haiti                            75
Portugal                         67
Taiwan                           65
Iran                             59
Greece                           49
Nicaragua                        49
Peru         

In [332]:
# Metodo para identificar el valor que aparece con mayor frecuencia dentro de la variable
df["native-country"].mode()

0    United-States
Name: native-country, dtype: object

In [326]:
# Metodo para identificar el número de valores únicos dentro de una variable
df["native-country"].nunique()

42

In [328]:
# Nombres de las categorías dentro de la variable
pd.unique(df["native-country"])

array(['United-States', 'Cuba', 'Jamaica', 'India', '?', 'Mexico',
       'South', 'Puerto-Rico', 'Honduras', 'England', 'Canada', 'Germany',
       'Iran', 'Philippines', 'Italy', 'Poland', 'Columbia', 'Cambodia',
       'Thailand', 'Ecuador', 'Laos', 'Taiwan', 'Haiti', 'Portugal',
       'Dominican-Republic', 'El-Salvador', 'France', 'Guatemala',
       'China', 'Japan', 'Yugoslavia', 'Peru',
       'Outlying-US(Guam-USVI-etc)', 'Scotland', 'Trinadad&Tobago',
       'Greece', 'Nicaragua', 'Vietnam', 'Hong', 'Ireland', 'Hungary',
       'Holand-Netherlands', nan], dtype=object)

In [330]:
# Metodo para contar los valores nulos
df["native-country"].isnull().sum()

274

##### income  
Es una variable cuantitativa, de tipo entero, cuenta con 48842 registros con 2 opciones de respuesta _>50K y <50K_, sin embargo, debido al punto, se evidencian 4 opciones. la variable no contiene datos nulos.

In [290]:
#información de la variable en cuanto a la cantidad de datos y tipo de dato.
df["income"].info()

<class 'pandas.core.series.Series'>
RangeIndex: 48842 entries, 0 to 48841
Series name: income
Non-Null Count  Dtype 
--------------  ----- 
48842 non-null  object
dtypes: object(1)
memory usage: 381.7+ KB


In [308]:
# Metodo para contar los valores nulos
df["income"].describe()

count     48842
unique        4
top       <=50K
freq      24720
Name: income, dtype: object

In [292]:
# Metodo para identificar el número de registros en la variable
df["income"].count()

48842

In [294]:
#Método con el cual se realiza el conteo de cada uno de los valores dentro de la variable
df["income"].value_counts()

income
<=50K     24720
<=50K.    12435
>50K       7841
>50K.      3846
Name: count, dtype: int64

In [296]:
# Metodo para identificar el número más grande dentro de la variable
df["income"].max()

'>50K.'

In [298]:
# Metodo para identificar el número más pequeño dentro de la variable
df["income"].min()

'<=50K'

In [300]:
# Metodo para identificar el valor que aparece con mayor frecuencia dentro de la variable
df["income"].mode()

0    <=50K
Name: income, dtype: object

In [302]:
# Metodo para identificar el número de valores únicos dentro de una variable
df["income"].nunique()

4

In [304]:
# Nombres de las categorías dentro de la variable
pd.unique(df["income"])

array(['<=50K', '>50K', '<=50K.', '>50K.'], dtype=object)

In [306]:
# Metodo para contar los valores nulos
df["income"].isnull().sum()

0

## <p><center> **ACTIVIDAD 2**</center></p>  
### Utilizar procesos con If para la limpieza de algunas variables de una Dataframe, para un análisis con datos de calidad.

In [31]:
d_f = pd.read_csv("titanic (1).csv")

In [33]:
#información de la variable en cuanto a la cantidad de datos y tipo de dato.
d_f.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 887 entries, 0 to 886
Data columns (total 8 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Survived                 887 non-null    int64  
 1   Pclass                   887 non-null    int64  
 2   Name                     887 non-null    object 
 3   Sex                      887 non-null    object 
 4   Age                      887 non-null    float64
 5   Siblings/Spouses Aboard  887 non-null    int64  
 6   Parents/Children Aboard  887 non-null    int64  
 7   Fare                     887 non-null    float64
dtypes: float64(2), int64(4), object(2)
memory usage: 55.6+ KB


In [35]:
#Identificación de cariables iniciales 
d_f.columns

Index(['Survived', 'Pclass', 'Name', 'Sex', 'Age', 'Siblings/Spouses Aboard',
       'Parents/Children Aboard', 'Fare'],
      dtype='object')

In [37]:
d_f.head()

Survived  Pclass                                               Name  \
0         0       3                             Mr. Owen Harris Braund   
1         1       1  Mrs. John Bradley (Florence Briggs Thayer) Cum...   
2         1       3                              Miss. Laina Heikkinen   
3         1       1        Mrs. Jacques Heath (Lily May Peel) Futrelle   
4         0       3                            Mr. William Henry Allen   

      Sex   Age  Siblings/Spouses Aboard  Parents/Children Aboard     Fare  
0    male  22.0                        1                        0   7.2500  
1  female  38.0                        1                        0  71.2833  
2  female  26.0                        0                        0   7.9250  
3  female  35.0                        1                        0  53.1000  
4    male  35.0                        0                        0   8.0500

In [47]:
# Metodo para identificar el número de valores únicos dentro de una variable
d_f["Survived"].nunique()

2

In [49]:
# Metodo para identificar el número de valores únicos dentro de una variable
d_f["Pclass"].nunique()

3

In [51]:
# Metodo para identificar el número de valores únicos dentro de una variable
d_f["Name"].nunique()

887

In [53]:
# Metodo para identificar el número de valores únicos dentro de una variable
d_f["Sex"].nunique()

2

In [57]:
# Metodo para identificar el número de valores únicos dentro de una variable
d_f["Siblings/Spouses Aboard"].nunique()

7

In [59]:
d_f["Siblings/Spouses Aboard"].value_counts()

Siblings/Spouses Aboard
0    604
1    209
2     28
4     18
3     16
8      7
5      5
Name: count, dtype: int64

In [61]:
# Metodo para identificar el número de valores únicos dentro de una variable
d_f["Parents/Children Aboard"].nunique()

7

In [63]:
d_f["Parents/Children Aboard"].value_counts()

Parents/Children Aboard
0    674
1    118
2     80
5      5
3      5
4      4
6      1
Name: count, dtype: int64

In [65]:
d_f["Fare"].nunique()

248